streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -1088,6 +1088,28 @@ public void shouldTriggerRecordingOfRocksDBMetricsIfRecordingLevelIsDebug() {
         }
     }
 
+    @Test
+    public void shouldGetClientSupplierFromConfigForConstructor() {
```

Comment:
Should we also test, that a supplier passed into the constructor overwrites the config?

Reply:
Sure.